### PR TITLE
Implement keyword for Monkhorst-Pack k-point sampling

### PR DESCRIPTION
**Pull request by awvwgk**
Related #846

### CODE REVIEW

File: src/dftbp/dftbplus/parser.F90 [hunk 1]
```diff
@@ -2744,6 +2745,24 @@ subroutine getEuclideanKSampling(tBadIntegratingKPoints, ctrl, node, geo)
       end if
       ctrl%nKPoint = size(ctrl%kPoint, dim=2)
 
+    case ("monkhorstpack")
+      call getChildValue(value1, "Repeat", coeffs)
+      call getChildValue(value1, "GammaCentered", gammaCentered, .false.)
+      if (gammaCentered) then
+        shifts = 0.0_dp
+      else
+        shifts = merge(0.5_dp, 0.0_dp, mod(coeffs, 2) == 0)
```
Comment (bhourahine) [inline src/dftbp/dftbplus/parser.F90:2754]: Not sure if this logic is generally correct, as while most unit cells should have a shift of .5,.5,.5 but hexagonal cases are need to be shifted by 0,0,.5 to retain the symmetry for even grids

**Review by bhourahine — COMMENTED** [review]
Might be an issue for the automatic shifts

File: src/dftbp/dftbplus/parser.F90 (hunk 1 above)
Comment (awvwgk) [reply src/dftbp/dftbplus/parser.F90:2754]: Means we have to detect the space group of the lattice first. Guess that is simple enough to do for the hexagonal case.

**Comment by bhourahine** [comment]
Not sure where they are getting this from, but the Vasp [documentation](https://www.vasp.at/wiki/index.php/KPOINTS) is also suggesting un-shifted grids that include Γ should be used for FCC and FC-orthorhomic (might be due to their symmetrization method). My understanding is the hexagonal case should only be shifted along c* (they say no shift at all).

**Comment by awvwgk** [comment]
Crystal symmetry might become difficult. We could use spglib, but maybe it is simple enough to implement at least the crystal systems ourselves to get some basic symmetry properties in. I tried the hexagonal one, but haven't tested it much yet.

**Comment by bhourahine** [comment]
Probably a good idea to have an alternative keyword to manually set the shift, just in case (another use case for this would be to make sure there is a sampling point on top of the Dirac cone in materials like graphene).

**Pull request closed without merging**